pkg/controllers/job/plugins/factory.go

```diff
@@ -29,10 +29,9 @@ func init() {
 	RegisterPluginBuilder("ssh", ssh.New)
 	RegisterPluginBuilder("env", env.New)
 	RegisterPluginBuilder("svc", svc.New)
+	RegisterPluginBuilder("mpi", mpi.New)
 }
 
-var pluginMutex sync.Mutex
```

Comment:
why remove this?

Reply:
unneccessary

Reply:
please keep this for now.

Reply:
For what

Reply:
How can you sure we will not use that later? I do not see any benefit, and it may introduce race condition later if anyone use it in multi thread scenario.

Reply:
I can not make sure.  But How does this differ with you add a new struct/interface that maybe used oneday?

Reply:
please keep this.